titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -608,16 +578,15 @@ private void publishContainerEvent(String taskId, TaskState taskState, String re
     }
 
     private void reconcileNodesAndPods() {
-        if (!mesosConfiguration.isReconcilerEnabled()) {
+        if (!mesosConfiguration.isReconcilerEnabled() || !nodeInformer.hasSynced() || !podInformer.hasSynced()) {
```

Comment:
Do we need `mesosConfiguration.isReconcilerEnabled()` check here? It is Mesos specific reconciliation logic.

Reply:
It is within Mesos because this is an integration that translates K8s into Mesos concepts.